### DISCUSSION TITLE
After upgrade to Heatwave 4 (version ) from WIX 3.14 build fails due to print button in dialog

### DISCUSSION THREAD

**Discussion by RAppels2022** [wixtoolset/issues#7257] [Questions] [answered]
I am using the standard WixUI_FeatureTree, in wix 3.14 there is no problem at all.
Now I understand that if the PushButton: 'Print' of Dialog: 'LicenseAgreementDlg' does not have an event defined in HeatWave 4 this is an error. How can I remove the 'Print' button of adjust that WixUI_FeatureTree dialog in the wxi file to fix this problem?
2>------ Rebuild All started: Project: Setup QueryReporter, Configuration: Release x86 ------
2>obj\x86\Release\Setup_Appelbol_QueryReporter 5.1.3.msi : error WIX0204: ICE17: PushButton: 'Print' of Dialog: 'LicenseAgreementDlg' does not have an event defined in the ControlEvent table. It is a 'Do Nothing' button.
2>Done building project "Setup QueryReporter.wixproj" -- FAILED.
Build has been canceled.
Projects build report:
  Status    | Project [Config|platform]
 -----------|---------------------------------------------------------------------------------------------------
  Succeeded | QueryReporter\QueryReporter.vbproj [Release|Any CPU]
  Failed    | Setup Abol QueryReporter\Setup  QueryReporter.wixproj [Release|x86]

**Comment by barnson** [comment, marked as answer]
https://github.com/wixtoolset/issues/issues/7249

**Reply by RAppels2022** [reply]
I have tried it in VS2019 (latest version 16.11.24 ) and VS2022 (latest version 17.5.1); both have Heatwave version 0.9.5 or in other words 4.0.0-rc3

**Reply by barnson** [reply]
It's not a bug.

**Reply by RAppels2022** [reply]
hahahaha well that is great to hear, I am so happy that it is not a bug! It just prevents my installer to build. So what would you call it? a major disruption? I can suppress  ICE 17 but I do not like that.

**Comment by RAppels2022** [comment]
I have read your response to the question #7249 and the difference is that I used your standard wix 3 "WixUI_FeatureTree dialog" where the problem comes from. I do not know if this is a bug or just an oversight in the conversion process or entirely something else. But I am asking for help and the remark "This is not a bug" is a laughable response sorry to say. Is the only solution to build my own "WixUI_FeatureTree dialog"? Then so be it.

**Reply by robmen** [reply]
Can you please provide more context about how you are "using WixUI_FeatureTree"? What is the exact .wxs code you are using in v3.14, and what are you using in v4.0? Our psychic-debugging skills are pretty good, but they tire when being constantly tested.

Also, please take a moment to use the `Preview` option in the GitHub text editor to see how your Markdown renders. Many of your questions and answers today have key information missing because you are not escaping the XML code properly.

**Reply by RAppels2022** [reply]
OK, I will try to be more specific and pay attention to the markdown in preview. Sorry for my initial reaction but the statement "It is not a bug" without any reason triggered me. I see what I can do to make everything more clear.

**Reply by robmen** [reply]
Sure. Note that vague/imprecise questions tend to trigger OSS maintainers because you implicitly push the burden of investigating your problem onto them. That tends to result in short answers that point you to enough information for you to spend the time and dig into the issue. If your next response shows you did no additional work with the provided information, then there is a very good chance that your problem will be (silently) dismissed with: 

> If it isn't important to them [the consumer] to help diagnose the problem, it isn't important to me [the OSS maintainer] to help root cause the issue.

Remember, you're not a customer here. You're just one of the thousands of _consumers_ using the project for free. [Customers](https://www.firegiant.com/services/) get our undivided time and attention.

**Reply by RAppels2022** [reply]
As I said before, if the cause is something completely different and I have to build my own replacement of the "WixUI_FeatureTree" to fix this problem, so be it! But I never complained that I was a customer, just to make sure we are all on the same page :-) I thought it was clear enough the error message that the build did generate. Obviously now it was not. Sorry for that oversight.

**Comment by RAppels2022** [comment]
I have 2 files zipped together, one is version 3.14 and the other is v4. As you can see I have not made any changes to the original WixUI_FeatureTree dialogs other then inset 2 additional windows to allow the user to put in there name and org and the path to the tsnames.ora. For the latter I have created a custom action dll when they browse for the path to that tnsnames.ora. 
[wix3_4.zip](https://github.com/wixtoolset/issues/files/10878848/wix3_4.zip)
This way I have no issues with the markdown ;)

**Comment by RAppels2022** [comment]
Took the solution from  #7249 and that works fully satisfying, marked the issue as answered. This topic can be closed.
